pywr/solvers/cython_glpk.pyx

```diff
@@ -677,6 +723,29 @@ cdef class CythonGLPKSolver(GLPKSolver):
         self.stats['objective_update'] += time.perf_counter() - t0
         t0 = time.perf_counter()
 
+        for agg_node in self.aggregated_with_factor_params:
+            factors = agg_node.get_factors(scenario_index)
+
+            # normalise factors
+            f0 = factors[0]
```

Comment:
Some of these temporary variables can be `cdef`'d. Check the annotated output by running `setup.py --annotate` and looking at the generated HTML file for this module.